src/DynamoCore/Core/TrustedLocatationsManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Dynamo.Configuration;
+using DynamoUtilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Dynamo.Core
+{
+    /// <summary>
+    /// Manages trusted locations for Dynamo
+    /// </summary>
+    public class TrustedLocatationsManager
+    {
+        /// <summary>
+        /// Use Lazy&lt;TrustedLocatationsManager&gt; to make sure the Singleton class is only initialized once
+        /// </summary>
+        internal static readonly Lazy<TrustedLocatationsManager>
+            lazy =
+            new Lazy<TrustedLocatationsManager>
+            (() => new TrustedLocatationsManager());
+
+        private PreferenceSettings settings;
+
+        /// <summary>
+        /// Dynamo's trusted locations.
+        /// Dynamo will load dlls from these locations without asking the user for consent.
```

Comment:
that comment is not correct - AFAIK trusted locations are for loading .dyn files, they have no affect on binaries, dyfs etc.